File: src/main/kotlin/com/chopeks/lexer/SquirrelDocLexer.java
package com.chopeks.lexer;

import com.chopeks._SquirrelLexer;
import com.intellij.lexer.FlexAdapter;
import com.intellij.lexer.MergingLexerAdapter;
import com.intellij.psi.tree.TokenSet;

import static com.chopeks.SquirrelTokenTypesSets.MULTI_LINE_COMMENT_BODY;

public class SquirrelDocLexer extends MergingLexerAdapter {

    public SquirrelDocLexer() {
        super(new FlexAdapter(new _SquirrelLexer()), TokenSet.create(MULTI_LINE_COMMENT_BODY));
    }
}

File: src/main/kotlin/com/chopeks/lexer/SquirrelDocLexer.kt
package com.chopeks.lexer

import com.chopeks.SquirrelTokenTypesSets.MULTI_LINE_COMMENT_BODY
import com.chopeks._SquirrelLexer
import com.intellij.lexer.FlexAdapter
import com.intellij.lexer.MergingLexerAdapter
import com.intellij.psi.tree.TokenSet

class SquirrelDocLexer : MergingLexerAdapter(FlexAdapter(_SquirrelLexer()), TokenSet.create(MULTI_LINE_COMMENT_BODY))